samples/23.react-live-canvas/src/useLiveCanvas.jsx

```diff
@@ -0,0 +1,43 @@
+import { InkingManager } from "@microsoft/live-share-canvas";
+import { useCallback, useEffect, useState } from "react";
+
+export const useLiveCanvas = (fluidContainer, hostingElement) => {
+  const [inkingManager, setInkingManager] = useState(undefined);
+
+  const handleBlackPenClick = useCallback(() => {
+    inkingManager.penBrush.color = { r: 0, g: 0, b: 0 }
+  }, [inkingManager])
+
+  const handleBluePenClick = useCallback(() => {
+    inkingManager.penBrush.color = { r: 0, g: 0, b: 255, a: 1 }
+  }, [inkingManager])
+  
+  const handleClearClick = useCallback(() => {
+    inkingManager.clear();
+  }, [inkingManager])
+
+  const startInkingManager = useCallback(async () => {
+    if (!fluidContainer || !hostingElement) {
+        return;
+    }
+    const inkingHost = hostingElement;
+    const inkManager = new InkingManager(inkingHost);
+    const liveCanvas = fluidContainer.initialObjects.liveCanvas;
+    await liveCanvas.initialize(inkManager);
+
+    // Activate the InkingManager so it starts handling pointer input
+    inkManager.activate();
+    setInkingManager(inkManager);
+  }, [fluidContainer, hostingElement]);
+
+  useEffect(() => {
+    startInkingManager();
+  }, [startInkingManager]);
+
+  return {
+    inkingManager,
+    handleBlackPenClick,

```

Comment:
Specifically - why would the developer call the `handleBlackPenClick` callback instead of calling `inkingManager.penBrush.color = { r: 0, g: 0, b: 0};` directly?

Developers tend to copy paste demo code, sometimes withoyut understanding what it actually does. Given this demo is basically our guidance for using `LiveCanvas`/`InkingManager` with React, I think we should change this `useLiveCanvas` method to only return the `InkingManager` instance.

Does that make sense?

Reply:
I can change the name to `setToBlackBrush` if that helps but these are all simple helpers to do common tasks abstracted away for folks who may need only basic functionality. For those who need advanced functions, these would serve as examples at the least and they could play with `inkingManager` as they like. For react, these callbacks provide good abstraction more than other FXs.

Reply:
Trying to add even more to my point. If the developer wants to change the tool to the laser pointer, what should they do? Call `inkingManager.tool = InkingTool.laserPointer;` or add a `useCallback` to the `useLiveCanvas` method and return that callback even it does exactly the same? To me the answer is obviously they should do the former - there is not value add in adding a new `useCallback`.

Reply:
as per discussion, moving these to the component